_posts/2024-04-29-dependabot-management.markdown

```diff
@@ -0,0 +1,206 @@
+
+## Common Issues Dependabot
+
+### Spam / Notification Fatigue
+
+<center>
+<div ><img src="/img/dependabot-management/notification_fatigue.png" width="400px"/><p>Image 1: Notification Fatigue</p></div>
+</center>
+
+Does this look familiar ? If so, my condolences comrade.  
+Dependabot when left unchecked can be a source of what feels like spam pull requests. Above we can see 49/51 open pull requests are related to dependencies updates and there are 34 security issues that need addressing. This is overwhelming and can lead to developers, understandably, ignoring these notifications. This can increase the attack surface of an application or organization.
+
+## AWS Code Artifact
+
+<center>
+<div ><img src="/img/dependabot-management/code_artifact_issue.png" width="400px"/><p>Image 2: AWS CodeArtifact</p></div>
+</center>
+
+Another common issue encountered with Dependabot is the inability to authenticate with AWS CodeArtifact at a organization level (to my knowledge). AWS Code Artifact is used for a variety of reasons including storing private packages and sharing them across your organization. In the context Dependabot updates, this can cause an issue as it is not supported out of the box. See Image 2: AWS CodeArtifact. Here, the original package is stored in AWS CodeArtifact and the updated package is stored in the public npm registry. This can cause issues with the integrity of the package and the security of the application.
+
+### Transient Dependencies
+
+Finally, another common issue with Dependabot is the handling of transient dependencies. Transient dependencies are dependencies that are not directly used by the application but are used by the dependencies of the application. A common situation where this becomes a problem is when you want have a large number of PRs, each containing a transient dependency upgrade.
+
+## Solutions
+
+### Notification Fatigue
+
+The first solution to notification fatigue is to limit the number and type of PRs raised by dependabot. Before raising those eyebrows please keep in mind this is exclusive of security updates. Security update PRs from Dependabot in fact can not be limited this config file. I am only talking about patch and minor upgrades being turned off.
```

Comment:
```suggestion
The first solution to notification fatigue is to limit the number and type of PRs raised by Dependabot. Before raising those eyebrows, please keep in mind this is exclusive of security updates. In fact, Security update PRs from Dependabot are not able to be limited by this config file. I am only talking about patch and minor upgrades being turned off.
```

I don't mind too much about the phrasing of the last bit, just read oddly in my mind